uberfire-backend/uberfire-backend-server/src/main/java/org/uberfire/backend/server/util/Paths.java

```diff
@@ -89,4 +90,35 @@ public static String readLockedBy(final Path path) {
     public static boolean isLock(final Path path) {
         return path.toURI().endsWith(PathFactory.LOCK_FILE_EXTENSION);
     }
+
+    /**
+     * Subtsring the second path from the first. This can be used to for example to substring the repository prefix from a Path.
+     * @param from The Path we want to substring
+     * @param what What we want to substring from the first Path
+     * @return The result. For example the File URI from the Repository root or from a submodule root.
+     */
+    public static String removePrefix(final Path from,
+                                      final Path what) {
+        PortablePreconditions.checkNotNull("From Path", from);
+        PortablePreconditions.checkNotNull("What Path", what);
```

Comment:
I would rename the parameters for clarity (naming them `wholePath`, `prefixToRemove` would make things clearer). Also I don't like that this method just "blindly" removes the substring of the length of prefix from the whole path without checking that the path actually has that prefix.

In JDK there are higher level apis for that like [Path#relativize](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html#relativize(java.nio.file.Path)) but not sure that we can use them here (since we have this "fork" of jdk nio :/ )

Reply:
The check for if the path actually starts with the prefix is a good idea. That is what was the original cause of this bug. One string started with git:// and another with default://. Although they are in theory the same thing, it is a good idea to check if the repository, team, branch and so on match.

Reply:
Team = Space, these names keep changing too fast.

Reply:
I take that back. I don't think we want a/../b here. So while relativize can be used in the method, the method name should be different.